watch_service.go

```diff
@@ -87,18 +87,19 @@ func (ws *watchService) watchRepo(ctx context.Context,
 	}
 
 	u := fmt.Sprintf("%vprojects/%v/repos/%v/contents%v", defaultPathPrefix, projectName, repoName, pathPattern)
-	ws.watchRequest(ctx, watchResult, u, lastKnownRevision, timeout)
-	return watchResult
+
+	return ws.watchRequest(ctx, u, lastKnownRevision, timeout)
 }
 
-func (ws *watchService) watchRequest(ctx context.Context, watchResult chan<- *WatchResult,
-	u, lastKnownRevision string, timeout time.Duration) {
+func (ws *watchService) watchRequest(
+	ctx context.Context,
+	u, lastKnownRevision string,
+	timeout time.Duration) *WatchResult {
```

Comment:
perhaps `time.Duration,\r\n)` for consistency?

Reply:
👍 Yeah, I would update it